### PR TITLE
Pictures Carousel

### LINKED ISSUE #265: Picture Carousel
This would be a component for displaying a series of photo cards on screen, with the ability to automatically cycle between them, example:

screen shot 2018-02-23 at 14 05 27

Reference: https://projects.invisionapp.com/d/main#/console/13404859/279933279/preview

### PR DESCRIPTION
**Pull request by fczuardi**
- closes #265

### CODE REVIEW

File: packages/carousel/src/styledComponents/Carousel.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import { colors } from "@crave/farmblocks-theme";
+
+const scale = 1.219;
+
+const Container = styled.div`
+  position: relative;
+  ul {
+    position: relative;
+    left: 50%;
+    margin-left: ${props => -props.itemConfig.width * scale / 2}px;
```
Comment (alcferreira) [inline packages/carousel/src/styledComponents/Carousel.js]: some parenthesis would be nice to make it better to read :P

`margin-left: ${props => - (props.itemConfig.width * scale / 2)}px;`

or even `margin-left: ${props => -1 * props.itemConfig.width * scale / 2}px;`

Also is it possible to leave some comments explaining these formulas? In the future we will forget why we used it this way

File: packages/carousel/src/styledComponents/Carousel.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    height: ${props =>
+      props.itemConfig.height * scale + props.itemConfig.margin * 2}px;
+    list-style: none;
+    padding: 0;
+    display: inline-flex;
+    align-items: center;
+
+    li {
+      margin: ${props => props.itemConfig.margin}px;
+      transition: all 2s;
```
Comment (alcferreira) [inline packages/carousel/src/styledComponents/Carousel.js]: can we receive `2s` from configs?

Imagine that `config.displayTime` is `1000ms`. We should be able to configure transition to use `0.5s` for instance

Comment (alcferreira) [reply packages/carousel/src/styledComponents/Carousel.js]: https://cravefoodservices.slack.com/files/U4KA0RLBD/F9VLURPCP/carousel.mov

File: packages/carousel/src/Carousel.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+      this.nextItem,
+      this.props.itemConfig.displayTime
+    );
+    return this.setState({ transitionInterval });
+  };
+
+  render() {
+    const { imageSet, itemConfig } = this.props;
+    return (
+      <Container activeItem={this.state.activeItem} itemConfig={itemConfig}>
```
Comment (alcferreira) [inline packages/carousel/src/Carousel.js]: We should merge the default config with `props.itemConfig`.

I did this and it loose the other configs:
```
<Carousel
          onChange={action("photo changed")}
          onEnd={action("end")}
          imageSet={imageSet}
          itemConfig={{
            displayTime: 3000
          }}
        />
```

Rendered this 
image

File: packages/carousel/src/Carousel.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    if (this.state.activeItem + 1 === this.props.imageSet.length) {
+      if (this.state.transitionInterval) {
+        window.clearInterval(this.state.transitionInterval);
+        this.props.onEnd();
+      }
+      return null;
+    }
+    const activeItem = this.state.activeItem + 1;
+    this.props.onChange(activeItem);
+    return this.setState({ activeItem });
```
[the remark this answers is not in the document]
Comment (fczuardi) [reply packages/carousel/src/Carousel.js]: I can set the state without returning it as well, no difference...

File: packages/carousel/src/Carousel.js [hunk 5]
```diff
@@ -18,35 +19,38 @@ const defaultConfig = {
...
-    return this.setState({ activeItem });
+    this.setState({ activeItem });
   };
 
   componentDidMount = () => {
-    const transitionInterval = window.setInterval(
+    this.transitionId = window.setInterval(
       this.nextItem,
-      this.props.itemConfig.displayTime
+      this.props.itemConfig.displayTime * 1000
```
Comment (alcferreira) [inline packages/carousel/src/Carousel.js]: 👍

File: packages/carousel/src/styledComponents/Carousel.js [hunk 6]
```diff
@@ -2,23 +2,26 @@ import styled from "styled-components";
 import { colors } from "@crave/farmblocks-theme";
 
 const scale = 1.219;
+const activeItemWidth = props => props.itemConfig.width * scale;
+const activeItemHeight = props => props.itemConfig.height * scale;
+const spaceBetweenItems = props => props.itemConfig.margin * 2;
+const stepDistance = props => props.itemConfig.width + spaceBetweenItems(props);
```
Comment (alcferreira) [inline packages/carousel/src/styledComponents/Carousel.js]: Beautiful props 😍

**Pull request merged**